File: CS50X_TS/PSET2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    if (strlen(text) == 0)
    {
        printf("Error: No text provided.\n");
        return 1;
    }

    float l = 0;
    float w = 1;
    float s = 0;
    int n = strlen(text);
    for (int i = 0; i < n; i++)
    {
        if (isalpha(text[i]))
        {
            l++;
        }
        if (text[i] == ' ')
        {
            w++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            s++;
        }
    }

    float L = 100 * (l / w);
    float S = 100 * (s / w);

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: CS50X_TS/PSET6/sentimental-readability/readability.py
from cs50 import get_string

# Get input text from user
input_text = get_string("Text: ")

# Initialize counters
letter_count = 0
for char in input_text:
    if char.isalpha():
        letter_count += 1

# Count words by splitting the text by spaces
word_count = len(input_text.split())

# Initialize sentence counter
sentence_count = 0
for char in input_text:
    if char in ['.', '!', '?']:
        sentence_count += 1

# Calculate average letters and sentences per 100 words
words_per_hundred = word_count / 100
avg_letters_per_100_words = letter_count / words_per_hundred
avg_sentences_per_100_words = sentence_count / words_per_hundred

# Calculate Coleman-Liau index
readability_index = round(0.0588 * avg_letters_per_100_words -
                          0.296 * avg_sentences_per_100_words - 15.8)

# Output the reading grade level
if readability_index < 1:
    print("Before Grade 1")
elif readability_index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {readability_index}")
